python/ray/air/execution/event_manager.py

```diff
@@ -0,0 +1,336 @@
+        TASKS: Only events relating to tasks or actor tasks will be awaited.
+        ACTORS: Only events relating to actor starts or stops will be awaited.
+
+    """
+
+    ALL = 0
+    TASKS = 1
+    ACTORS = 2
+
+
+@DeveloperAPI
+class RayEventManager:
+    """Management class for Ray actors, tasks, and actor tasks.
+
+    This class provides an event-based management interface for actors, tasks, and
+    actor tasks.
+
+    The manager can be used to start actors, stop actors, and schedule and
+    track task futures individually or on these actors.
+    The manager will then invoke callbacks related to the tracked entities.
+
+    For instance, when an actor is added with
+    :meth:`add_actor() <ray.air.execution.event_manager.RayEventManager.add_actor>`,
+    a :ref:`TrackedActor <ray.air.execution.tracked_actor.TrackedActor` object is
+    returned.
+    The :meth:`TrackedActor.on_start()
+    <ray.air.execution.tracked_actor.TrackedActor.on_start>`
+    method can then be used to specify a callback that is invoked once the actor
+    successfully started. The other callbacks relating to tracked actors
+    :meth:`TrackedActor.on_stop()
+    <ray.air.execution.tracked_actor.TrackedActor.on_stop>` and
+    :meth:`TrackedActor.on_error()
+    <ray.air.execution.tracked_actor.TrackedActor.on_error>`
+
+    Similarly, when scheduling an actor task using
+    :meth:`schedule_actor_task()
+    <ray.air.execution.event_manager.RayEventManager.schedule_actor_task>`,
+    a :ref:`TrackedActorTask <ray.air.execution.tracked_actor_task.TrackedActorTask`
+    object is returned.
+    The :meth:`TrackedActorTask.on_result()
+    <ray.air.execution.tracked_actor_task.TrackedActorTask.on_result>`
+    method can then be used to specify a callback that is invoked when the task
+    successfully resolved.
+    The :meth:`TrackedActorTask.on_error()
+    <ray.air.execution.tracked_actor_task.TrackedActorTask.on_error>`
+    method can then be used to specify a callback that is invoked when the task
+    fails.
+
+    Lastly, regular tasks (not on actors) can be tracked. When scheduling a task
+    using
+    :meth:`schedule_task()
+    <ray.air.execution.event_manager.RayEventManager.schedule_task>`,
+    a :ref:`TrackedTask <ray.air.execution.tracked_task.TrackedTask`
+    object is returned.
+    The :meth:`TrackedTask.on_result()
+    <ray.air.execution.tracked_task.TrackedTask.on_result>`
+    method can then be used to specify a callback that is invoked once the task
+    successfully resolved.
+    The :meth:`TrackedTask.on_error()
+    <ray.air.execution.tracked_task.TrackedTask.on_error>`
+    method can then be used to specify a callback that is invoked when the actor task
+    fails.
+
+    Args:
+        resource_manager: Resource manager used to request resources for the actors.
+
+    """
+
+    def __init__(self, resource_manager: ResourceManager):
+        self._resource_manager: ResourceManager = resource_manager
```

Comment:
Reading through this I now see why this is included here... One thing I'm wondering if we should expose a separate event type for `RESOURCES` scheduling as well, or if this can be encapsulated by Actor/Task errors entirely.

Reply:
resources is basically == actor start. In the current implementation it works well, so unless we identify the need to do this separately, let's keep it together. Adding that would be a simple change but I don't think it's needed.